code/_onclick/hud/alert.dm

```diff
@@ -626,7 +626,19 @@ so as to remain in compliance with the most up-to-date laws."
 	I.plane = FLOAT_PLANE
 	overlays += I
 
-/atom/movable/screen/alert/ghost/Click()
+/atom/movable/screen/alert/ghost/cryo/Click()
+	if(!..())
+		return
+	var/mob/living/carbon/human/target = usr
+	target.ghost()
+
+/atom/movable/screen/alert/ghost/xeno
+	name = "Ghost"
+	desc = "Would you like to ghost? You will be notified when your body is removed from the nest."
+	icon_state = "template"
+	timeout = 5 MINUTES // longer than any infection should be
+
+/atom/movable/screen/alert/ghost/xeno/Click()
 	if(!..())
 		return
```

Comment:
This means that this will run first, you should probably have a proc on `/atom/movable/screen/alert/ghost` that handles the ghosting functionality, i.e. `/proc/handle_ghosting()`
```
/atom/movable/screen/alert/ghost/cryo/Click()
	if(!..())
		return
	var/mob/living/carbon/human/target = usr
	target.ghost()
```